pkg/subscriptions/subscriptions-client.js

```diff
@@ -0,0 +1,416 @@
+                 * update status.
+                 */
+                deferred.resolve();
+                return;
+            }
+            requestUpdate();
+            var error = new Error(message);
+            deferred.reject(error);
+        });
+
+    promise = deferred.promise();
+    promise.cancel = function cancel() {
+        process.close("cancelled");
+        // we have no idea what the current state is
+        requestUpdate();
+    };
+
+    return promise;
+};
+
+client.unregisterSystem = function() {
+    var deferred = $.Deferred();
+
+    var args = ['subscription-manager', 'unregister'];
+
+    /* TODO DBus API doesn't deliver what we need, so we call subscription manager
+     * without translations and parse the output
+     */
+    var process = cockpit.spawn(args, {
+        directory: '/',
+        superuser: "require",
+        environ: ['LC_ALL=C'],
+        err: "out"
+    });
+
+    client.subscriptionStatus.status = "unregistering";
+    needRender();
+    var promise;
+    var buffer = '';
+    process
+        .input('')
+        .stream(function(text) {
+            buffer += text;
+        })
+        .done(function(output) {
+            deferred.resolve();
+        })
+        .fail(function(ex) {
+            if (ex.problem === "cancelled") {
+                deferred.reject(ex);
+                return;
+            }
+            var error = new Error(buffer.trim());
+            deferred.reject(error);
+            requestUpdate();
+        });
+
+    promise = deferred.promise();
+    promise.cancel = function cancel() {
+        process.close("cancelled");
+        // we have no idea what the current state is
+        requestUpdate();
+    };
+
+    return promise;
+};
+
+function statusUpdateFailed() {
+    if (updateTimeout !== null) {
+        client.subscriptionStatus.status = "not-found";
+        needRender();
+    }
+    updateTimeout = null;
+}
+
+/* request update via DBus
+ * possible status values: https://github.com/candlepin/subscription-manager/blob/30c3b52320c3e73ebd7435b4fc8b0b6319985d19/src/rhsm_icon/rhsm_icon.c#L98
+ * [ RHSM_VALID, RHSM_EXPIRED, RHSM_WARNING, RHN_CLASSIC, RHSM_PARTIALLY_VALID, RHSM_REGISTRATION_REQUIRED ]
+ */
+var subscriptionStatusValues = [
+    'RHSM_VALID',
+    'RHSM_EXPIRED',
+    'RHSM_WARNING',
+    'RHN_CLASSIC',
+    'RHSM_PARTIALLY_VALID',
+    'RHSM_REGISTRATION_REQUIRED'
+];
+var requestUpdate = function() {
+    service.call(
+        '/EntitlementStatus',
+        'com.redhat.SubscriptionManager.EntitlementStatus',
+        'check_status',
+        [])
+        .done(function(result) {
+            client.subscriptionStatus.serviceStatus = subscriptionStatusValues[result[0]];
+            updateTimeout = null;
+            client.getSubscriptionStatus();
+        })
+        .catch(function(ex) {
+            client.subscriptionStatus.serviceStatus = undefined;
+            updateTimeout = null;
```

Comment:
What about just ```window.clearTimeout(updateTimeout)```? And this probably should go into an always() or just go in the code below.